### PR TITLE
LMMS 1.2 Native Equalizer 2x Oversampling

### PR DESCRIPTION
**Pull request by DouglasDGI**
This adds a button that gives the LMMS 1.2 EQ 2x oversampling, preventing the curve distortion you get when you move the cutoff of any of the filters really high.

Here is a video to show the changes:
https://www.youtube.com/watch?v=9SH8IlLhXWs

Let me know if I messed anything up in the PR (still getting used to Github).

### CODE REVIEW

File: plugins/Eq/EqControlsDialog.cpp [hunk 1]
```diff
@@ -208,6 +208,14 @@ EqControlsDialog::EqControlsDialog( EqControls *controls ) :
 	hpBtnGrp->addButton( hp24Button );
 	hpBtnGrp->addButton( hp48Button );
 	hpBtnGrp->setModel( &m_controls->m_hpTypeModel,false);
+
+	//Oversample
+	PixmapButton * oversampleButton = new PixmapButton( this, NULL );
+	oversampleButton->setActiveGraphic( PLUGIN_NAME::getIconPixmap( "oversampleon" ) );
+	oversampleButton->setInactiveGraphic(  PLUGIN_NAME::getIconPixmap( "oversampleoff" ) );
+	oversampleButton->setCheckable(true);
```
Comment (PhysSong) [inline plugins/Eq/EqControlsDialog.cpp]: You should add `oversampleon.png` and `oversampleoff.png` to make the button usable.

**Comment by PhysSong** [comment]
@DouglasDGI I think you oversampled samples without any interpolation. It causes aliasing and may result in undesired behavior.
[4881.mmp.zip](https://github.com/LMMS/lmms/files/2955541/4881.mmp.zip)
To solve the problem, you should oversample the input signal correctly using e.g. `libsamplerate`. You can find some good examples by searching for `src_process` in this repository.

File: plugins/Eq/EqControlsDialog.cpp (hunk 1 above)
Comment (DouglasDGI) [reply plugins/Eq/EqControlsDialog.cpp]: Whoops!  Completely forgot to add those.

**Comment by PhysSong** [comment]
@DouglasDGI Do you think you can fix the aliasing issue? If you don't, can we close this for now and open a feature request for it?

**Comment by DouglasDGI** [comment]
I'll go ahead and close this, I was very new to DSP in general when I made this and didn't even understand how oversampling actually worked.
A feature request doesn't need to be opened, it's already listed in some Equalizer meta issue or something if I remember right...

**Comment by PhysSong** [comment]
Yes, I think #3605 mentions it.

**Pull request closed without merging**